revision/views.py

```diff
@@ -0,0 +1,26 @@
+from reversion.models import Version
+from reversion.views import RevisionMixin
+from reversion.errors import RegistrationError
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.exceptions import APIException
+from .serializers import RevisionSerializer
+
+
+class CustomRevisionMixin(RevisionMixin):
+
+    def get_versions(self):
+        instance = self.get_object()
+        try:
+            versions = Version.objects.get_for_object(instance)
+        except RegistrationError:
+            raise APIException(detail='model has not been registered for revision.')
+        current_version = versions[0]
```

Comment:
Ensure this ordering is explicitly set, as relying on default ordering can be error-prone. Also, consider edge cases where versions might be empty, which would raise an IndexError when accessing versions[0]
Moreover, maybe this answer will be useful https://stackoverflow.com/questions/9073830/previous-version-of-a-django-reversion-controlled-object

Reply:
Yes, ordering is set explicitly in Version model. Also the versions will never be empty because they are creating during Company creation.